src/stdatm/atmosphere_partials.py

```diff
@@ -0,0 +1,194 @@
+"""
+Implementation of International Standard Atmosphere with partial derivatives of state parameters
+with respect to altitude.
+"""
+#  This file is part of StdAtm
+#  Copyright (C) 2023 ONERA & ISAE-SUPAERO
+#  StdAtm is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from typing import Sequence, Union
+
+import numpy as np
+from scipy.constants import R, atmosphere, foot
+
+from .atmosphere import Atmosphere
+
+from .partials_state_parameters import (
+    compute_partial_temperature,
+    compute_partial_pressure,
+    compute_partial_density,
+    compute_partial_speed_of_sound,
+    compute_partial_dynamic_viscosity,
+    compute_partial_kinematic_viscosity,
+)
+
+AIR_MOLAR_MASS = 28.9647e-3
+AIR_GAS_CONSTANT = R / AIR_MOLAR_MASS
+SEA_LEVEL_PRESSURE = atmosphere
+SEA_LEVEL_TEMPERATURE = 288.15
+TROPOPAUSE = 11000
+
+
+class AtmospherePartials(Atmosphere):
+    """
+    Implementation of International Standard Atmosphere for troposphere and stratosphere with
+    derivatives of state parameters with respect to altitude.
+
+    Atmosphere properties and partials are provided in the same "shape" as provided
+    altitude:
+
+    - if altitude is given as a float, returned values will be floats
+    - if altitude is given as a sequence (list, 1D numpy array, ...), returned
+      values will be 1D numpy arrays
+    - if altitude is given as nD numpy array, returned values will be nD numpy
+      arrays
+
+    The AtmospherePartials class inherits from the Atmosphere class and thus retains its usages. It
+    however adds the computation of the partial derivatives of all state properties with respect to
+    the altitude.
+
+    Usage:
+
+    .. code-block::
+        >>> from stdatm import AtmospherePartials
```

Comment:
The empty line is needed.
```suggestion
    .. code-block::
    
        >>> from stdatm import AtmospherePartials
```